### PR TITLE
skip torchinductor test_data_type_propogation if C++ compiler is not available

### PR DESCRIPTION
**Pull request by williamwen42**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103920

This test is failing internally (https://www.internalfb.com/intern/test/844425024008760/).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78

### CODE REVIEW

**Review by yanboliang — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -6182,6 +6182,7 @@ def fn(a):
         opt_fn = torch._dynamo.optimize("inductor")(fn)
         same(fn(x), opt_fn(x))
 
+    @unittest.skipIf(not HAS_CPU, "requires C++ compiler")
```
Comment (jgong5) [inline test/inductor/test_torchinductor.py:6185]: @zhuhaozhe Weird that it has problems on other devices. The test should be general?

Comment (zhuhaozhe) [reply test/inductor/test_torchinductor.py:6185]: Hi, @williamwen42. May you help to give more info about how to reproduce?
I cannot open https://www.internalfb.com/login/?next=https%3A%2F%2Fwww.internalfb.com%2Fintern%2Ftest%2F844425024008760%2F.

From my understanding the only backend-related code is 
https://github.com/pytorch/pytorch/blob/a1118a9bfaf28a5b1727f3725922067b513ca1df/test/inductor/test_torchinductor.py#L6236
but I am not sure whether this should be an issue.

Comment (jansel) [reply test/inductor/test_torchinductor.py:6185]: This one is an issue with gcc not being installed on some machines.  Not sure there is something to fix.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**